governance/xc_admin/packages/xc_admin_frontend/hooks/useMultisig.ts

```diff
@@ -73,21 +75,31 @@ export const useMultisig = (): MultisigHookData => {
         })
       )
     } else {
-      setSquads(undefined)
+      setSquads(
+        new SquadsMesh({
+          connection,
+          wallet: new NodeWallet(new Keypair()),
+        })
+      )
     }
-  }, [wallet, urlsIndex, cluster, connection])
+  }, [wallet, connection])
+
+  useEffect(() => {
+    setReadOnlySquads(
+      new SquadsMesh({
+        connection,
+        wallet: new NodeWallet(new Keypair()),
+      })
+    )
+  }, [connection])
```

Comment:
This shouldn't be done with an effect & state, this should be done instead with `useMemo`

Reply:
fixed